File: VB/WinForms_MapControl_KeyColorColorizer/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraMap

Namespace WinForms_MapControl_KeyColorColorizer
    Partial Public Class Form1
        Inherits Form

        Private Const filepath As String = "..\..\Data\Hotel1.shp"

        Public Sub New()
            InitializeComponent()
            InitializeMap()
        End Sub

        Private Sub InitializeMap()
            ' Create a Map control.
            Dim map As New MapControl() With {.Dock = DockStyle.Fill, .CoordinateSystem = New CartesianMapCoordinateSystem(), .CenterPoint = New CartesianPoint(-100, -50), .MaxZoomLevel = 3, .MinZoomLevel = 1}
            Me.Controls.Add(map)

'            #Region "#CreateLayer"
            ' Create a vector layer and add it to the map.
            Dim hotelLayer As New VectorItemsLayer() With {.Data = CreateData(), .Colorizer = CreateColorizer()}
            map.Layers.Add(hotelLayer)
'            #End Region ' #CreateLayer

'            #Region "#AddLegend"
            ' Add a legend to the map.
            map.Legends.Add(New ColorListLegend() With {.Header = "Room Type", .Layer = hotelLayer})
'            #End Region ' #AddLegend
        End Sub

        #Region "#CreateData"
        ' Creates a shapefile data adapter.
        Private Function CreateData() As IMapDataAdapter
            Dim baseUri As New Uri(System.Reflection.Assembly.GetEntryAssembly().Location)

            Dim adapter As New ShapefileDataAdapter() With {.SourceCoordinateSystem = New CartesianSourceCoordinateSystem(), .FileUri = New Uri(baseUri, filepath)}

            Return adapter
        End Function
        #End Region ' #CreateData

        #Region "#CreateColorizer"
        ' Creates a key color colorizer.
        Private Function CreateColorizer() As MapColorizer
            Dim colorizer As New KeyColorColorizer() With {.ItemKeyProvider = New AttributeItemKeyProvider() With {.AttributeName = "CATEGORY"}, .PredefinedColorSchema = PredefinedColorSchema.Palette}

            colorizer.Keys.Add(New ColorizerKeyItem() With {.Key = 1, .Name = "Restaurant"})
            colorizer.Keys.Add(New ColorizerKeyItem() With {.Key = 2, .Name = "Business room"})
            colorizer.Keys.Add(New ColorizerKeyItem() With {.Key = 3, .Name = "Bathroom"})
            colorizer.Keys.Add(New ColorizerKeyItem() With {.Key = 4, .Name = "Living room"})
            colorizer.Keys.Add(New ColorizerKeyItem() With {.Key = 5, .Name = "Other"})
            colorizer.Keys.Add(New ColorizerKeyItem() With {.Key = 6, .Name = "Service room"})
            colorizer.Keys.Add(New ColorizerKeyItem() With {.Key = 7, .Name = "Pool"})
            colorizer.Keys.Add(New ColorizerKeyItem() With {.Key = 8, .Name = "Gym"})

            Return colorizer
        End Function
        #End Region ' #CreateColorizer
    End Class

End Namespace

File: CS/WinForms_MapControl_KeyColorColorizer/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraMap;

namespace WinForms_MapControl_KeyColorColorizer {
    public partial class Form1 : Form {
        const string filepath = "..\\..\\Data\\Hotel1.shp";
        
        public Form1() {
            InitializeComponent();
            InitializeMap();
        }

        private void InitializeMap() {
            // Create a Map control.
            MapControl map = new MapControl() {
                Dock = DockStyle.Fill,
                CoordinateSystem = new CartesianMapCoordinateSystem(),
                CenterPoint = new CartesianPoint(-100, -50),
                MaxZoomLevel = 3,
                MinZoomLevel = 1
            };
            this.Controls.Add(map);

            #region #CreateLayer
            // Create a vector layer and add it to the map.
            VectorItemsLayer hotelLayer = new VectorItemsLayer() {
                Data = CreateData(),
                Colorizer = CreateColorizer()
            };
            map.Layers.Add(hotelLayer);
            #endregion #CreateLayer

            #region #AddLegend
            // Add a legend to the map.
            map.Legends.Add(new ColorListLegend() {
                Header = "Room Type",
                Layer = hotelLayer
            });
            #endregion #AddLegend
        }

        #region #CreateData
        // Creates a shapefile data adapter.
        private IMapDataAdapter CreateData() {
            Uri baseUri = new Uri(System.Reflection.Assembly.GetEntryAssembly().Location);

            ShapefileDataAdapter adapter = new ShapefileDataAdapter() {
                SourceCoordinateSystem = new CartesianSourceCoordinateSystem(),
                FileUri = new Uri(baseUri, filepath)
            };
        
            return adapter;
        }
        #endregion #CreateData

        #region #CreateColorizer
        // Creates a key color colorizer.
        private MapColorizer CreateColorizer() {
            KeyColorColorizer colorizer = new KeyColorColorizer() {
                ItemKeyProvider = new AttributeItemKeyProvider() { AttributeName = "CATEGORY" },
                PredefinedColorSchema = PredefinedColorSchema.Palette
            };

            colorizer.Keys.Add(new ColorizerKeyItem() { Key = 1, Name = "Restaurant" });
            colorizer.Keys.Add(new ColorizerKeyItem() { Key = 2, Name = "Business room" });
            colorizer.Keys.Add(new ColorizerKeyItem() { Key = 3, Name = "Bathroom" });
            colorizer.Keys.Add(new ColorizerKeyItem() { Key = 4, Name = "Living room" });
            colorizer.Keys.Add(new ColorizerKeyItem() { Key = 5, Name = "Other" });
            colorizer.Keys.Add(new ColorizerKeyItem() { Key = 6, Name = "Service room" });
            colorizer.Keys.Add(new ColorizerKeyItem() { Key = 7, Name = "Pool" });
            colorizer.Keys.Add(new ColorizerKeyItem() { Key = 8, Name = "Gym" });

            return colorizer;
        }
        #endregion #CreateColorizer
    }

}
